python/tools/firmware-fingerprint.py

```diff
@@ -14,13 +15,19 @@ def firmware_fingerprint(filename, output):
     data = filename.read()
 
     try:
-        version, fw = firmware.parse(data)
+        # Unsigned production builds for Trezor T do not have valid code hashes.
+        # Use the internal module which recomputes them first.
+        if data[:4] == b"TRZV":
```

Comment:
T1 firmwares since 1.8.0 also start with `b"TRZV"`, we need different way how to distinguish between the two. Maybe @matejcik has an idea?

Reply:
Does it mean that https://github.com/trezor/trezor-firmware/blob/legacy/v1.9.3/python/src/trezorlib/firmware.py#L274 is wrong as well?

Reply:
Ah, sorry. I misread, I read "TRZF".

Reply:
i'd still prefer not having this information spread out through the codebase. Instead look at the `version` result from `firmware.parse` and if it's `TREZOR_T`, construct `firmware_headers.FirmwareImage(fw)` from the result.

Reply:
Fixed in b51380a6.